File: app/src/main/java/tech/thdev/app/presenter/MainContract.java
package tech.thdev.app.presenter;

import tech.thdev.app.adapter.model.SectionsPagerModel;
import tech.thdev.base.presenter.BasePresenter;
import tech.thdev.base.presenter.BaseView;

/**
 * Created by tae-hwan on 12/11/16.
 */

public interface MainContract {

    interface View extends BaseView {

        void updatePager();
    }

    interface Presenter extends BasePresenter<View> {

        void setSectionPagerModel(SectionsPagerModel sectionPagerModel);

        void loadSectionPagerItem();
    }
}

File: Android/2016-07-21-kotlin-example-01/app/src/main/kotlin/tech/thdev/kotlin_example_01/view/presenter/MainContract.kt
package tech.thdev.kotlin_example_01.view.presenter

import tech.thdev.kotlin_example_01.base.presenter.BasePresenter
import tech.thdev.kotlin_example_01.base.presenter.BaseView
import tech.thdev.kotlin_example_01.view.adapter.model.PhotoDataModel

/**
 * Created by Tae-hwan on 7/21/16.
 */
interface MainContract {

    interface View: BaseView<Presenter> {

        fun showFailLoad()

        fun refresh()

        fun showProgress()

        fun hideProgress()
    }

    interface Presenter: BasePresenter<View> {

        fun setDataModel(model: PhotoDataModel)

        fun loadPhotos(page: Int)
    }
}